infrastructure/src/main/java/org/corfudb/infrastructure/AbstractServer.java

```diff
@@ -38,14 +38,26 @@ public void sealServerWithEpoch(long epoch) {
 
     public abstract boolean isServerReadyToHandleMsg(CorfuMsg msg);
 
+    /**
+     * A stub that handlers can override to manage their threading, otherwise
+     * the requests will be executed on the IO threads
+     * @param msg
+     * @param ctx
+     * @param r
+     */
+    protected void processRequest(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+        getHandler().handle(msg, ctx, r);
```

Comment:
what if a handler is not present? Do we cover that case in underlying logic?

Reply:
Then it shouldn't compile because its abstract
`public abstract CorfuMsgHandler getHandler();`